opnieuw/retries.py

```diff
@@ -137,9 +139,30 @@ def __iter__(self) -> Iterator[Action]:
             )
 
 
-__retry_state_namespaces: Dict[Optional[str], Type[RetryState]] = defaultdict(
-    lambda: RetryState
-)
+__retry_state_namespaces: Dict[
+    Optional[str], ContextVar[Type[RetryState]]
+] = defaultdict(lambda: ContextVar("default_retry_state", default=RetryState))
```

Comment:
It might be nice for the name of this context variable to refer to opnieuw. It only really matters for debugging purposes, but considering this library might be imported by someone that does not know its internals, it would probably be good to help them find us.